### PR TITLE
repart: Add support for reading mkfs options from environment

### PR DESCRIPTION
**Pull request by daandemeyer**

### CODE REVIEW

File: src/shared/mkfs-util.c [hunk 1]
```diff
@@ -520,3 +520,22 @@ int make_filesystem(
...
         return 0;
 }
+
+int mkfs_options_from_env(const char *component, const char *fstype, char ***ret) {
+        _cleanup_(strv_freep) char **l = NULL;
+        const char *e;
+        char *n;
+
+        assert(fstype);
+
```
Comment (bluca) [inline src/shared/mkfs-util.c:532]: assert on ret and component (strjoina calls strlen() on it)

**Review by bluca — APPROVED** [review]
Nice, actually need this

**Comment by vcaputo** [comment]
> Nice, actually need this

Not trying to be glib here, but it's perhaps a little concerning/cause for reflection when a PR actually fulfilling a need is noteworthy...

**Comment by bluca** [comment]
> > Nice, actually need this
> 
> Not trying to be glib here, but it's perhaps a little concerning/cause for reflection when a PR actually fulfilling a need is noteworthy...

It might come as a surprise, but I do not personally have a use case for every single piece of code that is in this repository :-)

**Comment by poettering** [comment]
lgtm

**Comment by vcaputo** [comment]
> > > Nice, actually need this
> > 
> > 
> > Not trying to be glib here, but it's perhaps a little concerning/cause for reflection when a PR actually fulfilling a need is noteworthy...
> 
> It might come as a surprise, but I do not personally have a use case for every single piece of code that is in this repository :-)

I meant it more in the sense of reflecting on project scope creep, not an indictment of your (or my) narrow scope of use...

**Pull request merged**